apps/sms/views/shared/js/smil.js

```diff
@@ -365,9 +369,10 @@ window.SMIL = {
     if (!smil || attachmentsNotFound || !slides.length || smilMismatched) {
       // reset slides in the attachments not found case
       slides = Array(attachments.length);
-      attachments.forEach(SMIL_parseWithoutSMIL);
+      result = Promise.all(attachments.map(SMIL_parseWithoutSMIL));
     }
-    exitPoint();
+
+    return (result || Promise.resolve()).then(() => slides);
```

Comment:
I think we need separate bug to refactor this file including better error handling (I believe not mentored one, as it's not that trivial to make everything right :)).